### ISSUE TITLE
Question about open source.

### ISSUE THREAD

**Issue by Icecrest** [iChun/Morph#986]
To work on this on an independent fork, is iChunUtil required as a library in the workspace to develop?  I'd like to look into this and see if I can learn some stuff and maybe help with the development of the mod eventually. :)

ps. Love your stuff iChun

**Comment by iChun** [comment]
Yes, very much so. Morph uses a lot of functions only found in iChunUtil so
it is considered a hard dependency. Let me know if you have any issues
getting it working.

**Comment by Icecrest** [comment]
Forgive me, I'm not very experienced with this, I have only dabbled until I got completely stuck before.  When I try to build the iChunUtil to use a library in Morph, it is throwing an exception that it can't find the @SideOnly() annotation.  I think it is in an FML library that should be included, but I don't know how to include that.  I know you're a busy guy, so I am sorry for the pestering.  I'll include a screencap of the error message.  
![ichun_api_err01](https://user-images.githubusercontent.com/10534313/47483740-d32e1100-d7ee-11e8-9b89-a782d4bc51ac.png)

**Comment by Icecrest** [comment]
So I don't really understand the API situation, in the build.gradle it is trying to find a .jar, but from the Morph project, all there is is a package and a .java.  The other thing that is perplexing me is that I think half my problem is because I don't have a ForgeGradle set up, so the compiler can't find the forge source files.  The specific issue it's running into is the SideOnly which stacktraces back into the forge source files.  I'll try to figure out how to import ForgeGradle after cloning the project since I don't have the file on my computer.  I think I have to download it and then path it.

**Comment by iChun** [comment]
No you don't have to do that. Try running setupDecompWorkspace first.

**Comment by Icecrest** [comment]
Okay, when I ran setDecompWorkspace, it told me I have 105 Broken Acces Transformer Lines.  Any advice on how to deal with that?

**Comment by iChun** [comment]
It didn't stop the decompilation, right? Ignore it.

**Comment by iChun** [comment]
Try and build the project now

**Comment by Icecrest** [comment]
Tried to build, still failed the execution because of broken AccessTransformers.  Maybe I should try a clean fork of the project or something.

**Comment by Icecrest** [comment]
Nope, didn't help.  Used a local gradle wrapper I have installed, let things work themselves out then re-ran setupDecompWorspace and it failed and stopped at the 105 broken Access Transformers again.

**Comment by Icecrest** [comment]
Okay, progress is being made.  I got through a successful setupDecompWorkspace run, however, the gradlew build clean batch is still getting a failed execution due to the SideOnly annotation.  I'm not really sure how to fix that since all Forge imports should be findable at this point since the setupDecompWorkspace has been completed.

**Comment by Icecrest** [comment]
Don't know if this helps or hinders, but I'm using intelliJ IDEA as my IDE.  Also note, neither Minecraft files nor Forge files are appearing in the External Libraries portion of the window.

**Comment by iChun** [comment]
Are you running the idea task in gradle?

**Comment by Icecrest** [comment]
I had before, but just ran it again.  It added a lot of libraries, but the imports were still broken.  I went looking around for the files I needed, found them.  The path for Side and SideOnly were different (net.minecraftforge instead of cpw.mods).  However, the class in question (Api, from Morph) is still not grabbing those files properly even though it recognizes the proper path to it.
![importbug](https://user-images.githubusercontent.com/10534313/47674473-9b1f2900-db74-11e8-82e9-791c54cf1e6d.png)

**Comment by Icecrest** [comment]
Ok, it ironed things out.  Gave me a suggestion to add the jar to classpath, so I did and it worked properly.  I just attempted a new build, and now it is getting stopped at the Patron Tracker not having access to the MorphApi class.  I looked in both Morph and iChunUtil and I couldn't find any class by that name.

**Comment by iChun** [comment]
Are you on the right branch of morph? The master branch is outdated, use
the 1.12.2 branch.

**Comment by Icecrest** [comment]
Okay, it took some serious finagling, but it built finally.  I am not sure if I did the right thing, but I copied the entire api package and put it into me.ichun.mods.morph.*  I hope that was the right thing to do.  I think I have a handle from here, but I will probably be back once I learn that I have no clue what I am doing.  Thank you for being so patient and cool with all of this.

**Comment by Icecrest** [comment]
Okay, apparently we aren't quite out of the woods.  I placed iChunUtil-deobf.jar as a library for Morph to use, and it accepted it.  In the IDE, all classes can see and use any class that the library possesses.  The issue comes up when I try to run a build command and it fails over any time something from the *.ichunutil.common.core.* or *.ichunutil.client.* packages is used it threw an error that failed compilation.  I'm not sure what to do here.  I am on the correct branch of Morph, I have made sure my classpaths are correct and I have run both setupDecompWorkspace and idea in the Morph workspace.
